PhpStorm Protocol (Win)/run_editor.js

```diff
@@ -51,89 +32,58 @@ if (match) {
         }
     }
 
-    if (project !== '') {
-        editor += ' "%project%"';
-    }
-
-    editor += ' --line %line% "%file%"';
-
-    var command = editor.replace(/%line%/g, match[ 3 ])
-        .replace(/%file%/g, file)
-        .replace(/%project%/g, project)
-        .replace(/\//g, '\\');
-
-    shell.Exec(command);
-    shell.AppActivate(settings.window_title);
-}
-
-function isToolboxInstalled() {
-	var shell = new ActiveXObject('WScript.Shell'),
-        appDataLocal = shell.ExpandEnvironmentStrings("%localappdata%"),
-        toolboxDirectory = appDataLocal + '\\JetBrains\\Toolbox\\apps\\PhpStorm\\ch-0';
 
-	return (new ActiveXObject('Scripting.FileSystemObject')).FolderExists(toolboxDirectory);
-}
+    var fileSystemObj    = new ActiveXObject('Scripting.FileSystemObject'),
+        process_name     = 'phpstorm.exe',
+        dateLastModified = 0;
 
-function configureToolboxSettings(settings) {
-    var shell = new ActiveXObject('WScript.Shell'),
-        appDataLocal = shell.ExpandEnvironmentStrings("%localappdata%"),
-        toolboxDirectory = appDataLocal + '\\JetBrains\\Toolbox\\apps\\PhpStorm\\ch-0\\';
+        // see if is installed on a 64 path or 32 path...
+        for (var objEnum = new Enumerator(['C:\\Program Files (x86)\\JetBrains', 'C:\\Program Files\\JetBrains']); !objEnum.atEnd(); objEnum.moveNext()) {
 
-    // Reference the FileSystemObject
-    var fso = new ActiveXObject('Scripting.FileSystemObject');
+            var jetBrainsPath = objEnum.item();
 
-    // Reference the Text directory
-    var folder = fso.GetFolder(toolboxDirectory);
+            if (fileSystemObj.FolderExists(jetBrainsPath)) {
+                // get PHPStorm folder name from JetBrains path
+                for (var objEnum = new Enumerator(fileSystemObj.GetFolder(jetBrainsPath).SubFolders); !objEnum.atEnd(); objEnum.moveNext()) {
+                    var folderObject     = objEnum.item(),
+                        lastModifiedTime = new Date(folderObject.DateLastModified).getTime();
 
-    // Reference the File collection of the Text directory
-    var fileCollection = folder.SubFolders;
+                    if (lastModifiedTime > dateLastModified) {
+                        // see if is installed as 64 or 32
+                        if (fileSystemObj.FileExists(folderObject.Path + '\\bin\\phpstorm64.exe')) {
+                            process_name = 'phpstorm64.exe';
+                        }
 
-    var maxMajor = 0,
-        maxMinor = 0,
-        maxPatch = 0,
-        maxVersionFolder = "";
-    // Traverse through the fileCollection using the FOR loop
-    // read the maximum version from toolbox filesystem
-    for (var objEnum = new Enumerator(fileCollection); !objEnum.atEnd(); objEnum.moveNext()) {
-        var folderObject = ( objEnum.item() );
+                        var editor = folderObject.Path + '\\bin\\' + process_name;
 
```

Comment:
What was the purpose of replacing the whole EXE file detection code?

Reply:
Why keep something like this if I implemented automatic detection of the folder, the executable?

Reply:
Does it support all the use cases, that worked previously:

1. x86
2. x64
3. installation into the "Program Files..." folder (multiple version can be installed in parallel)
4. installation into JetBrains Toolbox folder (multiple version can be installed in parallel)

?